hibernate-core/src/main/java/org/hibernate/type/format/jakartajson/JsonBJsonFormatMapper.java

```diff
@@ -33,6 +34,10 @@ public JsonBJsonFormatMapper(Jsonb jsonb) {
 
 	@Override
 	public <T> T fromString(CharSequence charSequence, JavaType<T> javaType, WrapperOptions wrapperOptions) {
+		if ( charSequence == null || "null".contentEquals(charSequence) ) {
```

Comment:
The `charSequence` should never be `null`. Please move this check to the use-sites and ensure this method is never called with a `null` value.